test/unit/util/test_custom_templates.py

```diff
@@ -0,0 +1,59 @@
+"""Test custom template configuration and rendering."""
+
+from galaxy.util.custom_templates import template
+
+TEMPLATE_RELPATH = "mail/activation-email.txt"
+CONTEXT = {
+    "name": "Jane Doe",
+    "user_email": "jane@example.com",
+    "date": "01-01-2021",
+    "hostname": "example.galaxy.org",
+    "activation_url": "https://example.galaxy.org/fake-activation-link",
+    "terms_url": "https://example.galaxy.org/terms",
+    "contact_email": "contact@example.galaxy.org",
+    "instance_resource_url": "https://example.galaxy.org/resources",
+    "expiry_days": 7,
+}
+CUSTOM_TEMPLATE = """
+Here's my custom template. It starts with a comment section to let admins
+know how to use it.
+
+>>>>>> The actual template starts from the next line
+This is my custom template!
+
+Name: {{ name }}
+
+"""
+DEFAULT_TEMPLATE_HEADER_SUBSTRING = "The following variables are available"
+DEFAULT_TEMPLATE_OUTPUT_SUBSTRING = f"Hello {CONTEXT['name']}"
+CUSTOM_TEMPLATE_OUTPUT_SUBSTRING = f"Name: {CONTEXT['name']}"
+
+
+def test_it_can_render_a_default_template(tmp_path):
+    """It should fall back on codebase default."""
+    custom_templates_dir = tmp_path
+    output = template.render(
+        TEMPLATE_RELPATH,
+        CONTEXT,
+        custom_templates_dir,
+    )
+    # Make sure it's parsed out the doc header
+    assert DEFAULT_TEMPLATE_HEADER_SUBSTRING not in output
+    # Make sure the template content is correct
+    assert DEFAULT_TEMPLATE_OUTPUT_SUBSTRING in output
+
+
+def test_it_can_render_a_custom_template(tmp_path):
+    """It should use the custom template and ignore the default."""
+    custom_templates_dir = tmp_path
+    template_path = custom_templates_dir / TEMPLATE_RELPATH
+    template_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(template_path, "w") as f:
+        f.write(CUSTOM_TEMPLATE)
+    # It should find the custom template
+    output = template.render(
+        TEMPLATE_RELPATH,
+        CONTEXT,
+        custom_templates_dir,
+    )
+    assert CUSTOM_TEMPLATE_OUTPUT_SUBSTRING in output
```

Comment:
This is fine. But since you have full control over the text of the custom template here, you could just check for equality: `assert output == CUSTOM_TEMPLATE_OUTPUT`.

Reply:
Done